titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/common/action/TitusChangeAction.java

```diff
@@ -37,19 +39,25 @@
     private final String id;
     private final String name;
     private final String summary;
+    private final CallMetadata callMetadata;
 
     protected TitusChangeAction(TitusChangeAction delegate) {
         this.trigger = delegate.getTrigger();
         this.id = delegate.getId();
         this.name = delegate.getName();
         this.summary = delegate.getSummary();
+        this.callMetadata = delegate.getCallMetadata();
     }
 
-    public TitusChangeAction(Trigger trigger, String id, String name, String summary) {
+    public TitusChangeAction(Trigger trigger, String id, String name, String summary, CallMetadata callMetadata) {
         this.trigger = trigger;
         this.id = id;
         this.name = name;
         this.summary = summary;
+        // We assume that if call metadata is undefined, it comes from reconciler
+        this.callMetadata = callMetadata == null
```

Comment:
Can the reconciler set it explicitly such that there is no assumption and empty means a real issue?

Reply:
I am doing it as a safety measure. This should never happen, unless there is a bug in the code.

Reply:
Actually, maybe instead of assigning reconciler as a default, I should create a different one. Let me do that.